kclvm/tools/src/LSP/src/request.rs

```diff
@@ -128,7 +147,16 @@ pub(crate) fn handle_goto_definition(
 ) -> anyhow::Result<Option<lsp_types::GotoDefinitionResponse>> {
     let file = file_path_from_url(&params.text_document_position_params.text_document.uri)?;
     let path = from_lsp::abs_path(&params.text_document_position_params.text_document.uri)?;
-    let db = snapshot.get_db(&path.into())?;
+    let db = match snapshot.get_db(&path.clone().into())? {
```

Comment:
Consider adding KCL file filters on VFS.

Reply:
VFS only saves the mapping between FileId and sourcecode. When the file is changed, the sourcecode is compiled and the result is saved in the db. If it is not a valid kcl file, compilation fails and the corresponding compilation result will not be saved in the db. Therefore, if the request target is not a valid file, it will not be queried in the db. This problem has nothing to do with vfs or db, I think it should be filtered when processing requests